akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,23 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.foreach {
```

Comment:
ah yes, it should be the head. Maybe also rename "elements" to path-elements or something more eye-catching.